pkgs/top-level/all-packages.nix

```diff
@@ -5260,16 +5260,18 @@ in
 
   rtags = callPackage ../development/tools/rtags/default.nix {};
 
+  rustc = rustcStable;
+  rustcStable = callPackage ../development/compilers/rustc/stable.nix {};
+  rustcBeta = callPackage ../development/compilers/rustc/beta.nix {};
   rustcMaster = lowPrio (callPackage ../development/compilers/rustc/head.nix {});
-  rustc = callPackage ../development/compilers/rustc {};
 
   rustPlatform = rustStable;
-
   rustStable = recurseIntoAttrs (makeRustPlatform cargo rustStable);
+  rustBeta = lowPrio (recurseIntoAttrs (makeRustPlatform cargoUnstable rustBeta));
   rustUnstable = lowPrio (recurseIntoAttrs (makeRustPlatform cargoUnstable rustUnstable));
 
   # rust platform to build cargo itself (with cargoSnapshot)
-  rustCargoPlatform = makeRustPlatform (cargoSnapshot rustc) rustCargoPlatform;
+  rustCargoPlatform = makeRustPlatform (cargoSnapshot rustcStable) rustCargoPlatform;
   rustUnstableCargoPlatform = makeRustPlatform (cargoSnapshot rustcMaster) rustUnstableCargoPlatform;
```

Comment:
This is an opportunity to finally settle on a naming convention: `beta`, `unstable`, `head`, `master` are quite confusing. Maybe we should even remove the `rustUnstableCargoPlatform` and just keep `cargo`, `cargoUnstable` and all  `rustc`s and let users compose their own `unstable` platforms if needed.

Are there reasons not to do this?